### PR TITLE
Refactor Storage Handler

### LINKED ISSUE #91: Refactor Storage Handler
Currently, we are missing tests for `getListOfMembers` and `getListOfClubs`.
We also need to refactor `runDeleteMembershipTransaction` in order to not allow owners to leave their own clubs. Upon doing so, we must change /api/leave-club to account for new errors thrown due to exceptions when owners attempt to leave their own clubs. Additionally, getListOfMembers currently has two transaction calls, which is likely to cause errors upon testing. Thus, it must be refactored to not call `getMemberCount`, but rather return an exception if the list of members is empty (indicating there are none).

### PR DESCRIPTION
**Pull request by linamontes10-zz**
Add tests for `getListOfMembers` and `getListOfClubs`.
Refactor `runDeleteMembershipTransaction` in order to not allow owners to leave their own clubs.
Add tests to ensure that owners can not leave their club when calling `runDeleteMembershipTransaction`.
Change /api/leave-club to account for new errors thrown due to exceptions when owners attempt to leave their own clubs. Refactor `getListOfMembers` to not call `getMemberCount`, but rather return an exception if the list of members is empty (indicating there are none, which is impossible because at least there must be an owner).
Add tests for api/list-clubs to succesfully return an empty list of clubs if there are no clubs a user is not a member of (explore), or no clubs that a user is a member of (your-clubs).
Closes #91 .

### CODE REVIEW

File: server/src/main/java/com/google/coffeehouse/common/MembershipConstants.java [hunk 1]
```diff
@@ -22,6 +22,8 @@
   public static final int OWNER = 2;
 
   public static final String PERSON_ALREADY_IN_CLUB = "Person is already a member of club.";
+  public static final String NO_CLUBS = "No clubs that the person is a \"%s\" of.";
```
Comment (jimmycolema) [inline server/src/main/java/com/google/coffeehouse/common/MembershipConstants.java]: nit: it doesn't seem like this comment is called in the code; consider removing.

File: server/src/main/java/com/google/coffeehouse/storagehandler/StorageHandler.java [hunk 2]
```diff
@@ -186,25 +186,34 @@ public Void run(TransactionContext transaction) throws Exception {
...
   * Runs a transaction that deletes a membership (or ownership) from the database.
-  * This method checks if a person is already a member of a club by calling a helper function.
-  * If the person does exist, this method will buffer a single mutation that deletes the
-  * membership. Otherwise, it will throw an exception indicating that the person is
-  * already not a member of the club.
+  * This method checks if the person is already a member of a club by calling a helper function.
+  * This method also checks if the person is the owner of the club by calling a helper function.
+  * If the person is a member, and not the owner, this method will buffer a single mutation
+  * that deletes the membership. If the person is the owner of the club, it will throw an exception
+  * indicating that the owner can not leave their own club. If the person is not a member, it will
```
Comment (jimmycolema) [inline server/src/main/java/com/google/coffeehouse/storagehandler/StorageHandler.java]: nit: consider changing 'can not' to 'cannot'

File: server/src/main/java/com/google/coffeehouse/common/MembershipConstants.java (hunk 1 above)
Comment (linamontes10-zz) [reply server/src/main/java/com/google/coffeehouse/common/MembershipConstants.java]: Thanks for pointing this out, good catch Jimmy!

File: server/src/test/java/com/google/coffeehouse/storagehandler/StorageHandlerTest.java [hunk 3]
```diff
@@ -204,12 +208,183 @@ public void runAddAnyMembershipTypeTransaction_owner() throws Exception {
...
+      assertEquals(expected.get(i).getEmail(), actual.get(i).getEmail());
+      assertEquals(expected.get(i).getUserId(), actual.get(i).getUserId());
+      assertEquals(expected.get(i).getPronouns().get(), actual.get(i).getPronouns().get());
+    }
+  }
+
+  @Test
+  public void getListOfClubs_memberOfAll() throws Exception {
+    List<Club> expected = new ArrayList<Club>();
+    expected.add(StorageHandlerTestHelper.createTestClubObject("clubs1", /* contentWarning= */true));
```
Comment (jimmycolema) [inline server/src/test/java/com/google/coffeehouse/storagehandler/StorageHandlerTest.java]: nit: for this variable you're using camelCase, but for owner_id you're using snake case. consider using one throughout the code for stylistic consistency.

File: server/src/main/java/com/google/coffeehouse/storagehandler/StorageHandler.java [hunk 4]
```diff
@@ -186,25 +186,34 @@ public Void run(TransactionContext transaction) throws Exception {
...
+  * throw an exception indicating that the person is already not a member of the club.
   *
   * @param  dbClient    the database client
   * @param  userId      the user ID string used to perform the transaction
   * @param  clubId      the club ID string used to perform the transaction
   */
-  public static void runDeleteMembershipTransaction(DatabaseClient dbClient, String userId, String clubId) {
+  public static void runDeleteMembershipTransaction(DatabaseClient dbClient, String userId,
+                                                                             String clubId) {
```
Comment (KeatonTech) [inline server/src/main/java/com/google/coffeehouse/storagehandler/StorageHandler.java]: formatting nit: Cleaner way of wrapping args

```suggestion
  public static void runDeleteMembershipTransaction(
    DatabaseClient dbClient, 
    String userId, 
    String clubId
  ) {
```

**Review by KeatonTech — APPROVED** [review]

File: server/src/test/java/com/google/coffeehouse/storagehandler/StorageHandlerTest.java [hunk 5]
```diff
@@ -204,12 +208,183 @@ public void runAddAnyMembershipTypeTransaction_owner() throws Exception {
...
+  @Test
+  public void getListOfMembers_clubDoesNotExistInDb() throws Exception {
+    assertThrows(RuntimeException.class, () -> {
+      StorageHandler.getPerson(dbClient, "club");
+    });
+  }
+
+  @Test
+  public void getListOfMembers_none() throws Exception {
+    StorageHandlerTestHelper.insertClub("club", /* owner_id= */"owner");
```
Comment (JosephBushagour) [inline server/src/test/java/com/google/coffeehouse/storagehandler/StorageHandlerTest.java]:
```suggestion
    StorageHandlerTestHelper.insertClub("club", /* owner_id= */ "owner");
```
Parameter comments should have a space after the */, this should be changed throughout the file.

File: server/src/test/java/com/google/coffeehouse/storagehandler/StorageHandlerTest.java [hunk 6]
```diff
@@ -204,12 +208,183 @@ public void runAddAnyMembershipTypeTransaction_owner() throws Exception {
...
+
+  @Test
+  public void getListOfMembers_oneOwner() throws Exception {
+    List<Person> expected = new ArrayList<Person>();
+    expected.add(StorageHandlerTestHelper.createTestPersonObject("owner", /* pronouns= */true));
+
+    StorageHandlerTestHelper.insertPerson("owner");
+    StorageHandlerTestHelper.insertClub("club", /* owner_id= */"owner");
+    StorageHandlerTestHelper.insertMembership("owner", "club", MembershipConstants.OWNER);
+    List<Person> actual =  new ArrayList<Person>(StorageHandler.getListOfMembers(dbClient, "club"));
```
Comment (JosephBushagour) [inline server/src/test/java/com/google/coffeehouse/storagehandler/StorageHandlerTest.java]:
```suggestion
    List<Person> actual = new ArrayList<Person>(StorageHandler.getListOfMembers(dbClient, "club"));
```
Extra space, this occurs throughout the file.

File: server/src/test/java/com/google/coffeehouse/storagehandler/StorageHandlerTest.java [hunk 7]
```diff
@@ -204,12 +208,183 @@ public void runAddAnyMembershipTypeTransaction_owner() throws Exception {
...
+    StorageHandlerTestHelper.insertMembership("member", "clubs2", MembershipConstants.MEMBER);
+    StorageHandlerTestHelper.insertMembership("member", "clubs3", MembershipConstants.MEMBER);
+    StorageHandlerTestHelper.insertBook("book");
+    
+    List<Club> actual =  new ArrayList<Club>(StorageHandler.getListOfClubs(dbClient, "member",
+                                             MembershipConstants.MembershipStatus.MEMBER));
+
+    assertEquals(3, actual.size());
+    assertEquals(expected.size(), actual.size());
+    for (int i =0; i < actual.size(); i++) {
```
Comment (JosephBushagour) [inline server/src/test/java/com/google/coffeehouse/storagehandler/StorageHandlerTest.java]:
```suggestion
    for (int i = 0; i < actual.size(); i++) {
```
Missing space in for loop, this occurs twice in this file.

**Review by JosephBushagour — APPROVED** [review]
Looks good!
